### DISCUSSION TITLE
How to retain text cursor (caret) to its original position even after "setContents".

### DISCUSSION THREAD

**Discussion by Prithviraj-kf** [JiHong88/SunEditor#1099] [Q&A] [answered]
I have a situation where I need to do some regex and update the content on onTextChange.
but when i update content using "setContents" the cursor(caret) moves to the initial(first letter).

I want the caret to stay in the same position. Please help

**Comment by JiHong88** [comment, marked as answer]
You can use the [core.setRange](http://suneditor.com/sample/html/out/document-editor.html#setRange)
